### PR TITLE
feat(tab): add tab component

### PR DESCRIPTION
**Pull request by endv-bogdanb**

### CODE REVIEW

**Comment by dgonzalezr** [comment]
# ☁️ Nx Cloud Report

We didn't find any information for the current pull request with the commit 232a15653d149ade8b35aaf79a43c17d8e9ffecb.
Please verify you are running the latest version of the [NxCloud runner](https://www.npmjs.com/package/@nrwl/nx-cloud).

Check [the Nx Cloud Github Integration documentation](https://nx.dev/nx-cloud/set-up/github) for more information.

Alternatively, you can contact us at [cloud-support@nrwl.io](mailto:cloud-support@nrwl.io).

---

Sent with 💌 from [NxCloud.](https://nx.app)

File: packages/bee-q/src/components/tab-group/bq-tab-group.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        this.focusTabSibbling(target, 'forward');
+        break;
+      }
+      case 'ArrowLeft': {
+        this.focusTabSibbling(target, 'backward');
+        break;
+      }
+      default:
+    }
+  }
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/tab-group/bq-tab-group.tsx:128]: We should allow regular focus as well. I mean is good to change the selected Tab while pressing the Left/Right arrow key, but the user should be able to focus using the Tab key as well, without activating the focused Tab... Eg:

https://user-images.githubusercontent.com/328492/226837751-eca44e0f-0c0c-4b94-b3b2-38ad1df59b8c.mp4

File: packages/bee-q/src/components/tab/scss/bq-tab.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @apply py-[var(--bq-tab--pading-vertical-large)] px-[var(--bq-tab--pading-horizontal-large)] text-l;
+
+    ::slotted(bq-icon) {
+      --bq-icon--size: var(--bq-tab--icon-size-large) !important;
+    }
+  }
+}
+
+.bq-tab__underline {
+  @apply w-full border-x-0 border-t-0;
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/tab/scss/bq-tab.scss:42]: Using a `margin-top: 2px` on the underline will make the focus to looks better I guess:

CleanShot 2023-03-22 at 10 11 19

Comment (endv-bogdanb) [reply packages/bee-q/src/components/tab/scss/bq-tab.scss]: Done, thanks.

File: packages/bee-q/src/components/tab-group/bq-tab-group.tsx [hunk 3]
```diff
@@ -170,19 +185,37 @@ export class BqTabGroup {
...
+      }
+      bqTabElement.focusable(true);
+    });
+  };
+
+  private restoreTabsFocus = (): void => {
+    this.bqTabElements.forEach((bqTabElement) => {
+      if (bqTabElement.disabled || bqTabElement.active) {
+        return;
+      }
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/tab-group/bq-tab-group.tsx:203]:
```suggestion
      if (bqTabElement.disabled || bqTabElement.active) return;
```

File: packages/bee-q/src/components/tab/bq-tab.tsx [hunk 4]
```diff
@@ -116,6 +118,15 @@ export class BqTab {
     this.buttonElement?.blur();
   }
 
+  /**
+   * Sets tabindex on the native `<button>` HTML element used under the hood.
+   * This method is used inside `<bq-tab-group>` to make tab focusable after the active one is focused
+   */
+  @Method()
+  async focusable(value: boolean) {
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/tab/bq-tab.tsx:126]: Can we call this method: `enableFocus` instead?

File: packages/bee-q/src/components/tab-group/bq-tab-group.tsx (hunk 1 above)
Comment (endv-bogdanb) [reply packages/bee-q/src/components/tab-group/bq-tab-group.tsx]: That makes sense, updated.

File: packages/bee-q/src/components/tab-group/_storybook/bq-tab-group.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+const IconTemplate = (args) => {
+  return html`
+    <bq-tab-group
+      value="5"
+      .size=${args.size}
+      @bqChange=${args.bqChange}
+      @bqFocus=${args.bqFocus}
+      @bqBlur=${args.bqBlur}
+    >
+      <bq-tab tab-id="1"><bq-icon name="activity" slot="icon"></bq-icon>Tab</bq-tab>
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/tab-group/_storybook/bq-tab-group.stories.tsx:57]: I think the `activity` icon is no longer available, you should use `pulse` now 🙂 

CleanShot 2023-04-12 at 20 28 24

Comment (endv-bogdanb) [reply packages/bee-q/src/components/tab-group/_storybook/bq-tab-group.stories.tsx]: Done

File: packages/bee-q/src/components/tab/scss/bq-tab.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+@import './bq-tab-variables';
+
+:host {
+  @apply inline-block;
+}
+
+.bq-tab {
+  @apply relative flex cursor-pointer items-center justify-center disabled:cursor-not-allowed;
+  @apply text-m font-medium leading-large [&:not([disabled])]:hover:text-text-brand-hover;
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/tab/scss/bq-tab.scss:13]: Can we add some transition effects for hover and active states?

File: packages/bee-q/src/components/tab-group/bq-tab-group.tsx [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+      if (bqTabElement === currentTarget) {
+        target = getNextElement(elements, index, direction);
+      }
+    });
+
+    if (target) {
+      target.vFocus();
+      this.selectTab(target);
+    }
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/tab-group/bq-tab-group.tsx:159]:
```suggestion
    if (!target) return;
    
    target.vFocus();
    this.selectTab(target);
```

File: packages/bee-q/src/components/tab-group/bq-tab-group.tsx [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+  };
+
+  private makeTabsFocusable = (): void => {
+    this.bqTabElements.forEach((bqTabElement) => {
+      if (bqTabElement.disabled) {
+        return;
+      }
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/tab-group/bq-tab-group.tsx:166]:
```suggestion
      if (bqTabElement.disabled) return;
      
```

**Review by dgonzalezr — DISMISSED** [review]

File: packages/bee-q/src/components/tab-group/bq-tab-group.tsx (hunk 8 above)
Comment (endv-bogdanb) [reply packages/bee-q/src/components/tab-group/bq-tab-group.tsx]: Done

File: packages/bee-q/src/components/tab-group/bq-tab-group.tsx (hunk 7 above)
Comment (endv-bogdanb) [reply packages/bee-q/src/components/tab-group/bq-tab-group.tsx]: Done

**Review by dgonzalezr — APPROVED** [review]

**Pull request merged**